File: android/app/src/main/java/com/pearsonmedia/seddly/ui/theme/PremiumDesignSystem.kt
package com.pearsonmedia.seddly.ui.theme

import androidx.compose.animation.core.Spring
import androidx.compose.animation.core.SpringSpec
import androidx.compose.animation.core.spring
import androidx.compose.animation.core.tween
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * Premium design tokens (US-135) for Android — gradients, shadows, motion, and
 * expressive typography. Mirrors iOS PremiumDesignSystem.swift so both platforms
 * feel identical in the hand.
 */
object SeddlyPremium {

    // ── Gradients ──
    object Gradients {
        val hero: Brush = Brush.linearGradient(
            colors = listOf(
                Color(0xFF1A73E8),
                Color(0xFF5A3BD8),
                Color(0xFF7A2FC9)
            )
        )

        val proShimmer: Brush = Brush.horizontalGradient(
            colors = listOf(
                Color(0xFFF2C640),
                Color(0xFFFFE28C),
                Color(0xFFD89E28)
            )
        )

        fun status(color: Color): Brush = Brush.linearGradient(
            colors = listOf(
                color.copy(alpha = 0.20f),
                color.copy(alpha = 0.04f)
            )
        )

        @Composable
        fun cardSurface(): Brush = Brush.verticalGradient(
            colors = listOf(
                MaterialTheme.colorScheme.surface,
                MaterialTheme.colorScheme.surface.copy(alpha = 0.92f)
            )
        )
    }

    // ── Motion (spring presets matching iOS SeddlyMotion) ──
    object Motion {
        val snappy: SpringSpec<Float> = spring(
            stiffness = Spring.StiffnessHigh,
            dampingRatio = 0.72f
        )
        val smooth: SpringSpec<Float> = spring(
            stiffness = Spring.StiffnessMediumLow,
            dampingRatio = 0.82f
        )
        val celebratory: SpringSpec<Float> = spring(
            stiffness = Spring.StiffnessLow,
            dampingRatio = 0.55f
        )
        val easeSoft = tween<Float>(durationMillis = 280)
    }

    // ── Expressive Typography ──
    object DisplayFont {
        val hero = TextStyle(
            fontSize = 48.sp,
            fontWeight = FontWeight.Bold,
            fontFamily = FontFamily.Default
        )
        val display = TextStyle(
            fontSize = 34.sp,
            fontWeight = FontWeight.Bold
        )
        val accent = TextStyle(
            fontSize = 17.sp,
            fontWeight = FontWeight.SemiBold
        )
    }

    // ── Premium shapes ──
    object Shapes {
        val hero = RoundedCornerShape(20.dp)
        val sheet = RoundedCornerShape(28.dp)
    }
}

/** Applies a premium card treatment: continuous corners, soft shadow, hairline border. */
fun Modifier.premiumCard(): Modifier = this
    .shadow(
        elevation = 4.dp,
        shape = RoundedCornerShape(16.dp),
        ambientColor = Color.Black.copy(alpha = 0.05f),
        spotColor = Color.Black.copy(alpha = 0.12f)
    )
    .clip(RoundedCornerShape(16.dp))
    .padding(16.dp)

File: Seddly/Sources/Shared/PremiumDesignSystem.swift
import SwiftUI

// MARK: - Premium Design Tokens (US-135)
//
// Premium tier of the Seddly design system. Extends DesignSystem.swift with
// richer surfaces, gradients, shadows, motion tokens, and expressive typography
// to elevate the app above typical utility-style competitors. Every constant
// here is intentionally tuned so the app feels expensive in the hand without
// crossing into skeuomorphism.

// MARK: - Gradients

enum SeddlyGradient {
    /// Brand hero gradient — used for headers, onboarding, premium CTAs.
    static let hero = LinearGradient(
        colors: [
            Color.accentColor.opacity(0.95),
            Color.accentColor.opacity(0.65),
            Color(red: 0.35, green: 0.20, blue: 0.85).opacity(0.85)
        ],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )

    /// Subtle surface gradient for cards — one-stop darker toward bottom.
    static let cardSurface = LinearGradient(
        colors: [
            Color(.secondarySystemGroupedBackground),
            Color(.secondarySystemGroupedBackground).opacity(0.92)
        ],
        startPoint: .top,
        endPoint: .bottom
    )

    /// Gold shimmer for Pro feature locks.
    static let proShimmer = LinearGradient(
        colors: [
            Color(red: 0.95, green: 0.78, blue: 0.25),
            Color(red: 1.00, green: 0.90, blue: 0.55),
            Color(red: 0.85, green: 0.62, blue: 0.15)
        ],
        startPoint: .leading,
        endPoint: .trailing
    )

    /// Returns a subtle gradient tinted by a commitment status color.
    static func status(_ status: CommitmentStatus) -> LinearGradient {
        let base = SeddlyColors.status(status)
        return LinearGradient(
            colors: [base.opacity(0.18), base.opacity(0.04)],
            startPoint: .topLeading,
            endPoint: .bottomTrailing
        )
    }
}

// MARK: - Shadows (Depth)

struct SeddlyShadow {
    let color: Color
    let radius: CGFloat
    let x: CGFloat
    let y: CGFloat

    /// Soft ambient shadow used on cards.
    static let card = SeddlyShadow(
        color: Color.black.opacity(0.08),
        radius: 12,
        x: 0,
        y: 4
    )

    /// Raised shadow for active cards, sheets, floating elements.
    static let elevated = SeddlyShadow(
        color: Color.black.opacity(0.14),
        radius: 20,
        x: 0,
        y: 8
    )

    /// Subtle shadow used for pills and buttons.
    static let subtle = SeddlyShadow(
        color: Color.black.opacity(0.05),
        radius: 6,
        x: 0,
        y: 2
    )
}

extension View {
    /// Applies a Seddly depth shadow.
    func seddlyShadow(_ shadow: SeddlyShadow) -> some View {
        self.shadow(color: shadow.color, radius: shadow.radius, x: shadow.x, y: shadow.y)
    }
}

// MARK: - Motion (Spring Presets)

enum SeddlyMotion {
    /// Crisp spring — buttons, toggles, tap feedback.
    static let snappy: Animation = .spring(response: 0.28, dampingFraction: 0.72)
    /// Smooth spring — card transitions, sheet reveals.
    static let smooth: Animation = .spring(response: 0.45, dampingFraction: 0.82)
    /// Gentle bounce — celebratory moments (fulfill, new commitment).
    static let celebratory: Animation = .spring(response: 0.55, dampingFraction: 0.65)
    /// Soft ease — banner appearance, background shifts.
    static let easeSoft: Animation = .easeInOut(duration: 0.28)
}

// MARK: - Expressive Typography

enum SeddlyDisplayFont {
    /// Hero numeric display (accountability score, streak count).
    static let hero: Font = .system(size: 48, weight: .bold, design: .rounded)
    /// Large title for headers (Weekly Momentum, etc).
    static let display: Font = .system(.largeTitle, design: .rounded).weight(.bold)
    /// Subtitle headline for section accents.
    static let accent: Font = .system(.headline, design: .rounded).weight(.semibold)
    /// Numeric label with tabular figures for steady counters.
    static let tabularNumber: Font = .system(.title3, design: .rounded).monospacedDigit().weight(.semibold)
}

// MARK: - Premium Card Modifier

struct PremiumCardStyle: ViewModifier {
    var tintGradient: LinearGradient? = nil
    var padding: CGFloat = SeddlySpacing.xl
    var cornerRadius: CGFloat = SeddlyRadius.large
    var shadow: SeddlyShadow = .card

    func body(content: Content) -> some View {
        content
            .padding(padding)
            .background(
                ZStack {
                    SeddlyGradient.cardSurface
                    if let tint = tintGradient {
                        tint
                    }
                }
            )
            .clipShape(RoundedRectangle(cornerRadius: cornerRadius, style: .continuous))
            .overlay(
                RoundedRectangle(cornerRadius: cornerRadius, style: .continuous)
                    .strokeBorder(Color.white.opacity(0.06), lineWidth: 0.5)
            )
            .seddlyShadow(shadow)
    }
}

extension View {
    /// Premium card treatment: subtle gradient, continuous corners, hairline border, soft depth.
    func premiumCard(
        tint: LinearGradient? = nil,
        padding: CGFloat = SeddlySpacing.xl,
        cornerRadius: CGFloat = SeddlyRadius.large,
        shadow: SeddlyShadow = .card
    ) -> some View {
        modifier(PremiumCardStyle(
            tintGradient: tint,
            padding: padding,
            cornerRadius: cornerRadius,
            shadow: shadow
        ))
    }
}

// MARK: - Status Rail (vertical accent bar)

/// Thin gradient rail used as a status accent on cards.
struct StatusRail: View {
    let status: CommitmentStatus
    var width: CGFloat = 4

    var body: some View {
        RoundedRectangle(cornerRadius: width / 2, style: .continuous)
            .fill(
                LinearGradient(
                    colors: [
                        SeddlyColors.status(status),
                        SeddlyColors.status(status).opacity(0.55)
                    ],
                    startPoint: .top,
                    endPoint: .bottom
                )
            )
            .frame(width: width)
            .accessibilityHidden(true)
    }
}

// MARK: - Radii additions (premium sizes)

extension SeddlyRadius {
    /// 20pt - Hero cards, feature callouts.
    static let hero: CGFloat = 20
    /// 28pt - Sheet & modal corners on large content.
    static let sheet: CGFloat = 28
}
